src/misc/socket.cc

```diff
@@ -478,6 +478,14 @@ static ncclResult_t socketStartConnect(struct ncclSocket* sock) {
     }
     usleep(SLEEP_INT);
     return ncclSuccess;
+  } else if (errno == EHOSTUNREACH) {
+    if (++sock->noRouteRetries == RETRY_NO_ROUTE_TIMES) {
```

Comment:
In a healthy infra you not supposed to see this error. 
I would advocate the that error has to be reported on every retry to notify admins about the infra problem.

Reply:
I think that's a great idea.

Reply:
@newellz2 - thanks for putting this PR together

@shamisp - I'm the originator of the request for this. Please reach out to Sam Simcoe for details, nvidia support case is 00705873. High-level summary is that the SM takes maybe 6-8 seconds to program the fabric. In the NCCL failure scenarios I'm seeing the connect() attempt occurs during that 6-8 seconds, resulting in EHOSTUNREACH. When connect() is called at other times (when SM is not busy programming the fabric) then NCCL startup proceeds normally. A link flap that causes the SM to program the fabric is of course not ideal, but there are certainly other causes for the SM to program the fabric which are simply part of IB cluster daily life.